aten/src/ATen/core/SingletonSymNodeImpl.h

```diff
@@ -34,8 +35,8 @@ class TORCH_API SingletonSymNodeImpl : public SymNodeImpl {
  public:
   // CAUTION: you should probably not be constructing these directly; please
   // the higher-level API in python instead (TODO: actually introduce that).
-  explicit SingletonSymNodeImpl(int64_t val, int64_t coeff)
-      : val_(val), coeff_(coeff) {}
+  explicit SingletonSymNodeImpl(int64_t val, int64_t coeff, at::Tensor data, at::Tensor dummy, int64_t sum_offsets)
```

Comment:
can `sum_offsets` be symbolic? should we be storing SymInt here instead?

Reply:
It won't be symbolic here actually. After metafication, the symbolic singleton e.g. SymNode class in python will hold the cpp SingletonSymNode as hint, and THAT is the one holding the symbolic sum_offsets. The cpp SingletonSymNode in the hint will still hold concrete int64_t.